mycli/key_bindings.py

```diff
@@ -61,6 +61,45 @@ def _(event):
         else:
             b.start_completion(select_first=False)
 
+    @kb.add('c-x', 'p', filter=emacs_mode)
+    def _(event):
+        """
+        Prettify current statement into multiple lines.
+        """
+        _logger.debug('Detected <C-x p> key.')
+
+        b = event.app.current_buffer
+        cursorpos_relative = b.cursor_position / len(b.text)
+        pretty_text = mycli.handle_prettify_binding(b.text)
+        if len(pretty_text) > 0:
+            b.text = pretty_text
+            cursorpos_abs = int(round(cursorpos_relative * len(b.text)))
+            while 0 < cursorpos_abs < len(b.text) \
+                  and b.text[cursorpos_abs] in (' ', '\n'):
+                cursorpos_abs -= 1
+            b.cursor_position = min(cursorpos_abs, len(b.text))
+        # todo
+        # * informative feedback message on failure
```

Comment:
We could use the CLI toolbar at the bottom to show the error message. Here's an example: 

https://github.com/dbcli/mycli/blob/main/mycli/clitoolbar.py#L35

Reply:
resolved